Tickets/dm 7446

isullivan opened the pull request.

isullivan committed:
Add introduction text figures and equations

isullivan committed:
Add text describing notation and equations

isullivan committed:
Add Implementation text

isullivan committed:
Add text on end condition

isullivan committed:
Add decam and simulated images as examples

isullivan committed:
Add missing text

SimonKrughoff commented on index.rst:
```diff
@@ -1,51 +1,387 @@
+As with any medium, light entering Earth's atmosphere will refract.
+If the incident light arrives from any direction other than straight overhead, then refraction will bend its path towards an observer such that it appears to come from a location closer to zenith (see :numref:`fig-atmos_refraction`).
+While the bulk refraction is easily corrected during astrometric calibration, water vapor in the atmosphere increases the index of refraction for bluer light, which leads to an angular refraction that is wavelength-dependent.
+Since optical telescopes have a finite bandwidth, this means that a point-like source along the direction towards zenith, possibly across multiple pixels.
```
I think there is a verb missing in this sentence.

SimonKrughoff commented on index.rst:
```diff
@@ -1,51 +1,387 @@
-   Run: ``make html`` and ``open _build/html/index.html`` to preview your work.
-   See the README at https://github.com/lsst-sqre/lsst-technote-bootstrap or
-   this repo's README for more info.
+.. figure:: /_static/refraction.gif
```
I suspect this figure came from someplace on the web.  It would be good to add a reference I think.

SimonKrughoff commented on index.rst:
```diff
@@ -1,51 +1,387 @@
+==============================
 
-   A description of the algorithm to calculate a model of the true sky and forward model atmospheric effects to create matched templates.
+Because refraction is wavelength-dependent :eq:`eqn-refraction`, the smearing of astronomical sources will depend on their intrinsic spectrum. In practice, this appears as an elongation of the measured PSF in an image, and a jitter in the source location leading to mis-subtractions in difference imaging when looking for transient or variable sources.
```
It seems pedantic, but I've been trying to say "image differencing" instead of "difference imaging" simply because we don't actually image the difference.

SimonKrughoff commented on index.rst:
```diff
@@ -1,51 +1,387 @@
 
-   A description of the algorithm to calculate a model of the true sky and forward model atmospheric effects to create matched templates.
+Because refraction is wavelength-dependent :eq:`eqn-refraction`, the smearing of astronomical sources will depend on their intrinsic spectrum. In practice, this appears as an elongation of the measured PSF in an image, and a jitter in the source location leading to mis-subtractions in difference imaging when looking for transient or variable sources.
+This effect can be properly corrected when designing a telescope by incorporating an atmospheric dispersion corrector (ADC) in the optical path in front of the detector, but it cannot be removed in software once the photons have been collected.
```
I don't think it's too much detail to explain that LSST doesn't have an ADC and any why (note I don't know the official answer to why it doesn't).

SimonKrughoff commented on index.rst:
```diff
@@ -1,51 +1,387 @@
+Assuming we have no prior spectral information, the best initial guess is that all sub-bands have the same flux in all pixels.
+If all model planes are equal at the start, a good guess for the flux distribution within a sub-band is the standard co-add of the input images, divided by the number of model planes being used (since those will be summed).
+A proper inverse-variance weighting of the input images as part of the coaddition will help make the best estimate, and if there are many input images we could restrict the coaddition to use only those observed near zenith (with negligible DCR).
+An advantage of selecting the simple coadd as the starting point, is that the solution should immediately converge if the input data exhibits no actual DCR effects, such as *i*-band or zenith observations.
```
It's probably worth saying "redder bands (i and redder for LSST)", rather than specifically i-band.

SimonKrughoff commented on index.rst:
```diff
@@ -1,51 +1,387 @@
+As shown in :numref:`fig-sim_image` below, these images contain a moderately crowded field of stars (no galaxies) with Kolmogorov double-gaussian PSFs, realistic SEDs, photon shot noise, and no other effects other than DCR.
+In this example, the model is built using three frequency planes and eight input simulations of the field, with airmass ranging between 1.0 and 2.0 (*not* including the simulated observation shown in :numref:`fig-sim_image`).
+I use :eq:`eqn-basic_template` to build a DCR-matched template for the simulated observation (:numref:`fig-sim_template`), and subtract this template to make the difference image (:numref:`fig-sim_template_diff`).
+For comparison, in :numref:`fig-sim_image_diff` I subtract a second simulated image generated with the same field 10 degrees closer to zenith.
```
You should definitely say more about these diffs.  E.g. what are the residuals in Figure 6 (and what are their signal to noise).  The hamburgers in Figure 7 look strange to people that do this.  I know that PSF modeling will change how things look, but I think those residuals will worry people coming to this for the first time.

ebellm replied:
I echo Simon here--I think you want to emphasize that these are direct pixel subtractions and not Alard & Lupton-style image differences.

SimonKrughoff commented on index.rst:
```diff
@@ -1,51 +1,387 @@
+For a more rigorous test, I have also built DCR-matched templates for `DECam HiTS <https://arxiv.org/abs/1609.03567>`_ observations, which were calibrated and provided by Francisco Förster. 
+For these images I used the implementation outlined above using the simplified equation :eq:`eqn-iterative_sum`, despite the images having variable seeing.
+Because I have not yet implemented :eq:`eqn-psf_iterative_sum` using measured PSFs for each image, I have simply excluded observations with PSF FWHWs greater than 4 pixels (2.5 - 4 pixel widths are common). 
+However, it should be noted that the images with PSFs at the larger end of that range are not as well matched by their templates, so restricting the input images going into the model to those with good seeing may not be sufficient in the future.
```
Two things here.
1. I think you need to be clearer about what the differencing algorithm is.  Specifically this is *not* the stack algorithms, so describe what it *is*.
2. Those DECam images have cosmic rays in them.  It makes them look worse than they should.  I'm not sure what to do about that.  The stack pipelines will remove those, so you could try to use Meredith's stuff to reduce these images before diffing them.

SimonKrughoff commented on index.rst:
```diff
@@ -1,51 +1,387 @@
+=================
+
+
+Simulated source spectra
```
This section would be great to have.  I think it will really nail it down for people if you can show you are getting the right answer out.

SimonKrughoff reviewed (commented):
Overall it is looking good.  I think the level is generally good.  I added a couple comments where I thought it could be improved.

A broad comment is that we need to be clear about what algorithms are being used where and what changing algorithms will buy in the future.

isullivan committed:
Update DMTN build configuration

isullivan committed:
New simulated images

isullivan committed:
Expand introduction, add DCR skymodel and source spectra sections.

isullivan committed:
Merge branch 'master' of https://github.com/lsst-dm/dmtn-037

ebellm commented on index.rst:
```diff
@@ -1,51 +1,517 @@
+While it is often convenient for :math:`\overrightarrow{y_\alpha}` to have the same resolution and overall pixelization as :math:`\overrightarrow{s_i}`, in general even a non-gridded pixelization such as `HEALPix <https://healpix.jpl.nasa.gov>`_ could be used.
+
+The matrix :math:`B_{i\alpha}` encodes the transformation due to DCR of model plane :math:`\alpha` to image :math:`i`, and the reverse transformation is written as :math:`B_{\alpha i}^\star`.
+Since the sub-bands have a narrow bandwidth, the effect of DCR is a uniform shift of all pixels, so:
```
How narrow do the sub-bands need to be for this approximation to hold?

isullivan replied:
I've added a new section describing finite bandwidth effects, and how I deal with them.

ebellm commented on index.rst:
```diff
@@ -1,51 +1,517 @@
+.. figure:: /_static/simulations/DCR_dipole_plot.png
+   :name: fig-dcr_dipoles
+
+   Plot of the number of false detections for simulated images as a function of airmass, for different templates.
```
Can you comment on the number of false detections expected from pure noise, given the size of your images?

isullivan replied:
I believe there should be ~1 false detection due to noise. I've added a comment.

ebellm reviewed (changes requested):
A handful of comments and questions.

isullivan committed:
Add section on the finite bandwidth of sub-bands and high airmass observations.

isullivan committed:
Add clarifying comments about direct pixel subtractions.

isullivan committed:
Update plots with results using finite bandpass.

isullivan committed:
Add comment on number of false detections from noise.

isullivan commented:
I have added a new section on how I deal with finite bandwidth effects, and updated the figures to use the results from using finite bandwidth. You should see a few improvements!
